src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Profile.cshtml.cs

```diff
@@ -0,0 +1,47 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using APIViewWeb.Models;
+using APIViewWeb.Repositories;
+using System.Threading.Tasks;
+
+namespace APIViewWeb.Pages.Assemblies
+{
+    public class ProfileModel : PageModel
+    {
+        private readonly UserProfileManager _manager;
+        public readonly UserPreferenceCache _preferenceCache;
+
+        public UserProfileModel userProfile;
+        public ProfileModel(UserProfileManager manager, UserPreferenceCache preferenceCache)
+        {
+            _manager = manager;
+            _preferenceCache = preferenceCache;
+        }
+
+        public async Task<IActionResult> OnGetAsync(string UserName)
```

Comment:
Why do we need UserName here? Can't we always use username from `User`? I think accepting username can open up a possibility to update other user's profile by calling GET method. It's always better to rely on the username in User whenever possible instead of accepting it from incoming request to make a secured entry point.

Reply:
This allows viewing another's profile, which is checked and ensures that the edit page only displays if username is null (goes to the user's profile), or they are accessing their own profile